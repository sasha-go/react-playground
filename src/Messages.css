.messages {
  display: block;
  position: relative;
  padding: 0 12px;
  height: 22px;
  line-height: 26px;
  font-size: 11px;
  font-weight: bold;
  color: #666;
  text-decoration: none;
  text-shadow: 0 1px white;
  background: #FAFAFA;
  width: 100px;
  box-shadow: inset 0 0 3px rgba(0, 0, 0, 0.1);
  margin: 15px auto;
}

.messages .unread_count {
  display: block;
  position: absolute;
  top: -10px;
  right: 3px;
  line-height: 14px;
  height: 14px;
  padding: 0 5px;
  font-family: Arial, sans-serif;
  font-size: 9px;
  color: white;
  text-shadow: 0 1px rgba(0, 0, 0, 0.25);
  border: 1px solid;
  border-radius: 10px;
  box-shadow: inset 0 1px rgba(255, 255, 255, 0.3), 0 1px 1px rgba(0, 0, 0, 0.08);
  background: #67c1ef;
  border-color: #30aae9;
}