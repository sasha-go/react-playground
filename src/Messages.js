import React from 'react';
import './Messages.css';

export default function Messages(props) {
  const badge = +props.unread 
    ? <div className="unread_count badge">{props.unread}</div>
    : null;
  return (
    <div className="messages">
      {props.name} 
      {badge}
    </div>  
  );
}