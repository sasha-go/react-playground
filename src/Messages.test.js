import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Messages from './Messages';

describe('Messages component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Messages />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no unreads', () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={0}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});