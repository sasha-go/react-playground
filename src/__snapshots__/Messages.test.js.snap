// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Messages component renders the UI as expected with no unreads 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
